action.yml

```
@@ -124,6 +128,15 @@ runs:
         echo $CID
         echo "✅ Merkleized path: \`${{ inputs.path-to-deploy }}\` into CAR file with root CID \`$CID\`" >> $GITHUB_STEP_SUMMARY
 
+    - name: Upload CAR file as artifact
+      if: ${{ inputs.upload-car-artifact != 'false' }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ github.event.pull_request.head.sha || github.sha }}-${{ steps.merkleize.outputs.cid }}.car
```

- prefixing with repo name makes it more user-friendly when saving CARs produced by different repo 
- i think `github.sha` is always resent? it can be trimmed to 7 characters  to match github ui and still be useful.
- produces `org-repo_1234567_cid.car`

```suggestion
        name: ${{ github.repository | replace('/', '-') }}_${{ github.sha | substring(0, 7) }}_${{ steps.merkleize.outputs.cid }}.car
```

---

> prefixing with repo name makes it more user-friendly when saving CARs produced by different repo

👍

> i think `github.sha` is always present? it can be trimmed to 7 characters to match github ui and still be useful.

In pull requests:
- `github.sha` points to the merge commit SHA - this is a temporary commit that would result from merging the PR branch into the base branch
- `github.event.pull_request.head.sha` points to the latest commit SHA on the PR branch - this is the actual commit that contains the changes being proposed

 The problem with the suggested change is that there is no `[replace` function in GithHub expressions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#functions)

---

should be fixed in https://github.com/ipfs/ipfs-deploy-action/pull/19/commits/6b8f42e9e539bc5ab0624183eb31cd0383e59f5c